azurerm/resource_arm_app_configuration.go

```diff
@@ -0,0 +1,320 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	appconf "github.com/Azure/azure-sdk-for-go/services/appconfiguration/mgmt/2019-10-01/appconfiguration"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAppConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppConfigurationCreate,
+		Read:   resourceArmAppConfigurationRead,
+		Update: resourceArmAppConfigurationUpdate,
+		Delete: resourceArmAppConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		SchemaVersion: 2,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAppConfigurationName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "free",
+				ValidateFunc: validation.StringInSlice([]string{
+					"free",
+					"standard",
+				}, false),
+			},
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_read_key": {
+				Type:      schema.TypeMap,
```

Comment:
Hey @mbfrahry, thanks for the review. Would you like to keep the differentiation between primary and secondary keys? If so which way would you prefer:

1. use two different list, one for primary, one for secondary keys
2. add an attribute that tells if this key is primary or secondary

Feel free to suggest another solution.

Reply:
After thinking about this I'm not sure if using a list makes it easier for users to access the keys.

Right now the interface for accessing the primary read key secret looks like this: `azurerem_app_configuration.appconf.primary_read_key.secret`

How should the interface look like after switching to a list?

Reply:
I believe all the logic you have can stay around dictating what key is being referred to. I'm just suggesting we change what object we're packaging it into since a Map is used for when we don't know the keys. Switching to a List would like:

```
"primary_write_key": {
	Type:      schema.TypeList,
        MaxItems: 1,
	Computed:  true,
	Elem: &schema.Schema{
		Schema: map[string]*schema.Schema{
                          "secret": {
				Type:     schema.TypeString,
				Computed: true,
				Sensitive: true,
			},
			"connection_string": {
				Type:     schema.TypeString,
				Computed: true,
				Sensitive: true,
                        },
                },
	},
},
```
I'd argue that `id` isn't necessary unless we're passing it along to another resource.

Reply:
From my understand the id is required to connect to the App configuration as well as the secret. At least the connection string contains both of them.

Reply:
Makes sense to me! Let's keep `id` in